api/session.go

```diff
@@ -179,6 +180,8 @@ func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, d
 			// Attempt a session destroy
 			s.Destroy(id, q)
 			return nil
+		case <-ctx.Done():
```

Comment:
Context cancellation is also supposed to release any associated resources. For a canonical example -  HttpRequest's context cancel closes client initiated tcp connections. We'd rather not leave dangling sessions here, so its best to destroy the session if its been cancelled.